ambry-frontend/src/main/java/com/github/ambry/frontend/AmbrySecurityService.java

```diff
@@ -314,6 +320,12 @@ private void setGetBlobResponseHeaders(BlobInfo blobInfo, GetBlobOptions options
         restResponseChannel.setHeader("Content-Disposition", "attachment");
       }
     }
+    if (blobProperties.getContentEncoding() != null) {
+      restResponseChannel.setHeader(Headers.AMBRY_CONTENT_ENCODING, blobProperties.getContentEncoding());
+    }
+    if (blobProperties.getFilename() != null) {
+      restResponseChannel.setHeader(Headers.AMBRY_FILENAME, blobProperties.getFilename());
```

Comment:
In our LinkedIn SecurityService, we use the filename to set the value of the Content-Disposition. Not sure if we want to port that behavior here.

Reply:
Removed.